src/3d/qgswindow3dengine.cpp

```diff
@@ -0,0 +1,61 @@
+#include "qgswindow3dengine.h"
+
+#include <Qt3DRender/QRenderCapture>
+#include <Qt3DExtras/Qt3DWindow>
+#include <Qt3DExtras/QForwardRenderer>
+
+
+QgsWindow3DEngine::QgsWindow3DEngine()
+{
+  mWindow3D = new Qt3DExtras::Qt3DWindow;
```

Comment:
Leaked?

Reply:
I think you only have to delete the window, it takes ownership of the entities and cleans them up. But probably a good idea to verify that.

Reply:
The idea is that the 3D window gets embedded into a QWidget container which takes ownership of the QWindow (according to QWidget::createWindowContainer() documentation) - so it should not leak..